src/components/3d_editor/advanced_geometry/python_transformation/PythonTransformation.tsx

```diff
@@ -0,0 +1,175 @@
+        super(props);
+        this.state = {
+            materials: props.materials,
+            selectedMaterials: [props.materials[0]],
+            isLoading: true,
+            isRunning: false,
+            pyodide: null,
+            pythonCode: "",
+            pythonOutput: "",
+        };
+    }
+
+    componentDidUpdate(prevProps: PythonTransformationProps) {
+        const { materials } = this.props;
+        if (prevProps.materials !== materials) {
+            // eslint-disable-next-line react/no-did-update-set-state
+            this.setState({ materials });
+        }
+    }
+
+    onPyodideLoad = (pyodideInstance: any) => {
+        this.setState({ pyodide: pyodideInstance, isLoading: false }, async () => {
+            // redirecting stdout for print and errors per https://pyodide.org/en/stable/usage/streams.html
+            pyodideInstance.setStdout({
+                batched: (text: string) => this.redirectPyodideStdout(text),
+            });
+        });
+    };
+
+    redirectPyodideStdout = (text: string) => {
+        this.setState((prevState) => ({
+            pythonOutput: prevState.pythonOutput + text + "\n",
+        }));
+    };
+
+    runPythonCode = async () => {
+        const { pyodide, pythonCode } = this.state;
+        this.setState({ pythonOutput: "" });
+
+        try {
+            await pyodide.runPythonAsync(pythonCode);
+        } catch (error: any) {
+            this.setState({ pythonOutput: error.message });
+        }
+    };
+
+    handleRun = async () => {
+        this.setState({ isRunning: true });
+        try {
+            await this.runPythonCode();
+        } catch (error: any) {
+            NPMsAlert.error(error.message);
+        } finally {
+            this.setState({ isRunning: false });
+        }
+    };
+
+    handleSubmit = () => {
+        // TODO: return newly created materials in the next task
+        // eslint-disable-next-line react/destructuring-assignment
+        this.props.onSubmit(this.state.selectedMaterials);
+    };
+
+    render() {
+        const { isLoading, isRunning, pythonCode, pythonOutput, materials, selectedMaterials } =
+            this.state;
+        const { show, onHide } = this.props;
+
+        return (
+            <Dialog
+                id="python-transformation-dialog"
+                open={show}
+                onClose={onHide}
+                fullWidth
+                maxWidth="xl"
+                onSubmit={this.handleSubmit}
+                title="Python Transformation"
+                isSubmitButtonDisabled={isLoading || isRunning}
+                scroll="body"
+            >
+                <PyodideLoader onLoad={this.onPyodideLoad} triggerLoad={show} />
+                <Box mt={theme.spacing(1)} p={`0 ${theme.spacing(3)}`}>
+                    <Grid container spacing={theme.spacing(2)}>
+                        <Grid item xs={12} sm={12} md={5}>
+                            <MaterialsSelector
```

Comment:
1. This should go inside DialogContent - see https://github.com/Exabyte-io/materials-designer/pull/157
2. PythonCode Display should be part of the grid layout too
3. We should reuse MaterialsSelector in DefaultImportModal

Reply:
Awesome suggestion. Somehow now with updated dialogs and using Grid, the scroll behaves as expected where expected 😃 
![image](https://github.com/Exabyte-io/materials-designer/assets/79542055/5a001fb0-1959-4c35-92ca-802e9aeae7e9)